File: examples/CBuilder/CODEGENU.cpp
//---------------------------------------------------------------------------
#include <vcl\vcl.h>
#pragma hdrstop

#include "CodeGenU.h"
//---------------------------------------------------------------------------
#pragma link "OnGuard"
#pragma resource "*.dfm"
TCodeGenFrm *CodeGenFrm;
//---------------------------------------------------------------------------
__fastcall TCodeGenFrm::TCodeGenFrm(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TCodeGenFrm::CopySbClick(TObject *Sender)
{
  int OldSelStart = CodeEd->SelStart;
  CodeEd->SelStart = 0;
  CodeEd->SelLength = MaxInt;
  CodeEd->CopyToClipboard();
  CodeEd->SelStart = OldSelStart;
  CodeEd->SelLength = 0;
}
//---------------------------------------------------------------------------
void __fastcall TCodeGenFrm::GenerateBtnClick(TObject *Sender)
{
  TKey CKey = { 0xE5,0x8F,0x84,0xD6,0x92,0xC9,
  	0xA4,0xD8,0x1A,0xFA,0x6F,0x8D,0xAB,0xFC,0xDF,0xB4 };
	TCode Work;
  // set default key to use
  OgMakeCodes1->SetKey(CKey);
  if (OgMakeCodes1->Execute()) {
    // get the code
    OgMakeCodes1->GetCode(Work);
    // display the code
    CodeEd->Text = BufferToHex(&Work, sizeof(TCode));
  }
}
//---------------------------------------------------------------------------

File: examples/Delphi/CODEGENU.pas
unit CodeGenU;

interface

uses
  {$IFDEF Win32} Windows, {$ELSE} WinTypes, WinProcs, {$ENDIF}
  Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls, Buttons, OnGuard, OgUtil;

type
  TCodeGenFrm = class(TForm)
    GenerateBtn: TButton;
    CodeEd: TEdit;
    Label2: TLabel;
    CloseBtn: TBitBtn;
    CopySb: TSpeedButton;
    OgMakeCodes1: TOgMakeCodes;
    procedure CopySbClick(Sender: TObject);
    procedure GenerateBtnClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  CodeGenFrm: TCodeGenFrm;

implementation

{$R *.DFM}

procedure TCodeGenFrm.CopySbClick(Sender: TObject);
var
  OldSelStart : Integer;
begin
  OldSelStart := CodeEd.SelStart;
  CodeEd.SelStart := 0;
  CodeEd.SelLength := MaxInt;
  CodeEd.CopyToClipboard;
  CodeEd.SelStart := OldSelStart;
  CodeEd.SelLength := 0;
end;

procedure TCodeGenFrm.GenerateBtnClick(Sender: TObject);
const
  CKey : TKey = ($E5,$8F,$84,$D6,$92,$C9,$A4,$D8,$1A,$FA,$6F,$8D,$AB,$FC,$DF,$B4);
var
  Work : TCode;
begin
  {set default key to use}
  OgMakeCodes1.SetKey(CKey);
  if OgMakeCodes1.Execute then begin
    {get the code}
    OgMakeCodes1.GetCode(Work);
    {display the code}
    CodeEd.Text := OgUtil.BufferToHex(Work, SizeOf(Work));
  end;
end;

end.
